src/Microsoft.Identity.Client/ClientApplicationBase.WithBuilders.cs

```diff
@@ -66,57 +65,44 @@ internal ClientApplicationBase(ApplicationConfiguration config)
         }
 
         async Task<AuthenticationResult> IClientApplicationBaseExecutor.ExecuteAsync(
-            IAcquireTokenSilentParameters silentParameters,
+            AcquireTokenCommonParameters commonParameters,
+            AcquireTokenSilentParameters silentParameters,
             CancellationToken cancellationToken)
         {
-            var authorityInstance = string.IsNullOrWhiteSpace(silentParameters.AuthorityOverride)
-                                        ? GetAuthority(silentParameters.Account)
-                                        : Instance.Authority.CreateAuthority(ServiceBundle, silentParameters.AuthorityOverride);
+            var customAuthority = string.IsNullOrWhiteSpace(commonParameters.AuthorityOverride)
+                                      ? GetAuthority(silentParameters.Account)
+                                      : Instance.Authority.CreateAuthority(ServiceBundle, commonParameters.AuthorityOverride);
 
-            ApiEvent.ApiIds apiId = string.IsNullOrWhiteSpace(silentParameters.AuthorityOverride)
-                                        ? ApiEvent.ApiIds.AcquireTokenSilentWithoutAuthority
-                                        : ApiEvent.ApiIds.AcquireTokenSilentWithAuthority;
+            var requestParameters = CreateRequestParameters(commonParameters, UserTokenCacheInternal, customAuthority);
 
-            var handler = new SilentRequest(
-                ServiceBundle,
-                CreateRequestParameters(silentParameters, UserTokenCacheInternal, authorityInstance),
-                apiId,
-                silentParameters.ForceRefresh);
+            requestParameters.Account = silentParameters.Account;
```

Comment:
if there is more than one account, is there a way to specific which account to pick up?

Reply:
Not sure I understand your question.  This is the account specified by the builder.

---
In reply to: [248924092](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/798#discussion_r248924092) [](ancestors = 248924092)

Reply:
i see. thanks.

---
In reply to: [249126974](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/798#discussion_r249126974) [](ancestors = 249126974,248924092)